### PR TITLE
Polymorphize auction bid flashes

### PR DESCRIPTION
**Pull request by jessieay**
* Quack quack

Minor:

* Remove unused js and related specs

### CODE REVIEW

File: app/presenters/bid_status_flash_factory.rb [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  def lowest_user_bid
+    user_bids.order(amount: :asc).first
+  end
+
+  def user_bids
+    auction.bids.where(bidder: user)
+  end
+end
+
+class OverWithBids
```
Comment (jessieay) [inline app/presenters/bid_status_flash_factory.rb]: trying out putting super small classes all in one file when they are only serving one small function -- what do people think? I like not needing to open many files to see what's going on but I also find it weird to have many classes per file

**Comment by adelevie** [comment]
Ready? Or still WIP?

File: app/presenters/bid_status_flash_factory.rb (hunk 1 above)
Comment (harrisj) [inline app/presenters/bid_status_flash_factory.rb]: I don't think Rails' autoloader likes that :(

File: app/view_models/auction_show_view_model.rb [hunk 2]
```diff
@@ -31,11 +31,7 @@ def rules_link_text
   end
 
   def rules_path
-    if auction.type == 'single_bid'
-      '/auction/rules/single-bid'
-    else
-      '/auctions/rules/multi-bid'
-    end
+    "/auctions/rules/#{auction.type.dasherize}"
```
Comment (harrisj) [inline app/view_models/auction_show_view_model.rb]: I would like to eventually rename single-bid and multi-bid to some better names

Comment (jessieay) [inline app/view_models/auction_show_view_model.rb]: totally!

**Comment by jessieay** [comment]
@adelevie ready!

File: app/presenters/bid_status_flash_factory.rb (hunk 1 above)
Comment (jessieay) [inline app/presenters/bid_status_flash_factory.rb]: it is working here since we are only referencing the classes within this file -- what do you think?

Comment (harrisj) [inline app/presenters/bid_status_flash_factory.rb]: Oh okay then

Comment (adelevie) [inline app/presenters/bid_status_flash_factory.rb]: I'm ok with multiple class files ... Maybe stuck into their own folder. We could also maybe start sticking README.md files in these folders to document/explain the API of some of the files in that folder.

**Comment by adelevie** [comment]
LGTM

**Pull request merged**